File: tooling/intellij/stratos-language/src/main/kotlin/com/araizen/stratos/files/StratosFileType.java
package com.araizen.stratos.files;

import com.intellij.openapi.fileTypes.LanguageFileType;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import javax.swing.*;

public class StratosFileType extends LanguageFileType {
    public static final StratosFileType INSTANCE = new StratosFileType();

    private StratosFileType() {
        super(StratosLanguage.INSTANCE);
    }

    @NotNull
    @Override
    public String getName() {
        return "Stratos File";
    }

    @NotNull
    @Override
    public String getDescription() {
        return "Stratos language file";
    }

    @NotNull
    @Override
    public String getDefaultExtension() {
        return "st";
    }

    @Nullable
    @Override
    public Icon getIcon() {
        return StratosFileIcons.FILE;
    }
}

File: tooling/intellij/stratos-language/src/main/kotlin/com/araizen/stratos/files/StratosFileType.kt
package com.araizen.stratos.files

import com.araizen.stratos.files.StratosFileType.Companion.INSTANCE
import com.intellij.openapi.fileTypes.LanguageFileType
import javax.swing.Icon

class StratosFileType private constructor() : LanguageFileType(StratosLanguage.INSTANCE) {
    override fun getName(): String {
        return "Stratos File"
    }

    override fun getDescription(): String {
        return "Stratos language file"
    }

    override fun getDefaultExtension(): String {
        return "st"
    }

    override fun getIcon(): Icon? {
        return StratosFileIcons.FILE
    }

    companion object {
        val INSTANCE = StratosFileType()
    }
}